src/main.cpp

```diff
@@ -1189,16 +1189,16 @@ int handleCmdLine()
 
     if(CommandLine::has("--numkarts", &n) ||CommandLine::has("-k", &n))
     {
-        UserConfigParams::m_num_karts = n;
-        if(UserConfigParams::m_num_karts > stk_config->m_max_karts)
+        UserConfigParams::m_default_num_karts = n;
+        if(UserConfigParams::m_default_num_karts > stk_config->m_max_karts)
         {
             Log::warn("main", "Number of karts reset to maximum number %d.",
                       stk_config->m_max_karts);
-            UserConfigParams::m_num_karts = stk_config->m_max_karts;
+            UserConfigParams::m_default_num_karts = stk_config->m_max_karts;
         }
-        race_manager->setNumKarts( UserConfigParams::m_num_karts );
+        race_manager->setNumKarts( UserConfigParams::m_default_num_karts );
```

Comment:
Perhaps remove the saving to userconfigparams::m_default_num_karts, and just call setnumkarts (after ensuring its valid)

Reply:
Could be worth also appending the minormode to the end of the major mode, thoughts?